### PR TITLE
Feature: platform specific access key

### PR DESCRIPTION
**Pull request by danymill**

### CODE REVIEW

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js [hunk 1]
```diff
@@ -66,36 +66,42 @@ const AccessKeyModalController = function AccessKeyModalController(
...
+                    const languages = data.languages || [];
+                    const providerConfigUrl = data.providerConfigUrl || '';
 
-                languages.sort((a, b) => a.order - b.order);
-                vm.languages = languages;
+                    languages.sort((a, b) => a.order - b.order);
+                    vm.languages = languages;
 
-                const languagesPromises = vm.languages.length ? loadLanguagesSnippets() : [];
+                    const languagesPromises = vm.languages.length ? loadAllSnippets(): [];
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: space before `:`

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js [hunk 2]
```diff
@@ -110,20 +116,33 @@ const AccessKeyModalController = function AccessKeyModalController(
...
+                    const snippetUrl = langConfig.snippetURLs[key];
+                    const snippet = await loadLanguageSnippet(snippetUrl);
 
-                    return langConfig;
+                    return ({ [key]: snippet });
                 });
-        });
+            const snippets = await $q.all(snippetLoaders);
+
+            langConfig.snippets = snippets.reduce((accum, item) => ({...accum, ...item}), {});
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: spaces before/after brackets

**Review by dzmitrymishyn — APPROVED** [review]

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js (hunk 1 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js (hunk 2 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: fixed

**Pull request merged**